log/log_root_signer.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"github.com/golang/glog"
+
+	"github.com/google/trillian/crypto"
+	"github.com/google/trillian/crypto/sigpb"
+	"github.com/google/trillian"
+)
+
+// CreateAndSignLogRoot creates a to-be-SignedLogRoot proto given the func arguments,
+// signs and returns it.
+func CreateAndSignLogRoot(signer *crypto.Signer, rootHash []byte, tsNanos, treeSize, logID, revision int64) (*trillian.SignedLogRoot, error) {
```

Comment:
As it stands, the method duplicates SignedLogRoot's constructor just to call CreateLogRootSignature() with it. I think removing this method and exposing just CreateLogRootSignature leads to more flexible functions and results in better readability in the caller.